projects/vaults/contracts/vaults/Symbiotic/InceptionVault_S.sol

```diff
@@ -371,11 +424,40 @@ contract InceptionVault_S is MellowHandler, IInceptionVault_S {
         return _claimerWithdrawals[claimer].amount;
     }
 
-    function maxRedeem(
-        address account
-    ) external view returns (uint256 maxShares) {
-        return
-            convertToAssets(IERC20(address(inceptionToken)).balanceOf(account));
+    function decimals() public view virtual returns (uint8) {
+        return IERC20Metadata(address(inceptionToken)).decimals();
+    }
+
+    function maxDeposit(address) public view virtual returns (uint256) {
+        return type(uint256).max;
+    }
+
+    function maxMint(address) public view virtual returns (uint256) {
+        return type(uint256).max;
+    }
+
+    function maxWithdraw(address owner) public view virtual returns (uint256) {
```

Comment:
Let’s add some notes indicating that the extra functions were specifically added for ERC-4626 compatibility. Additionally, include comments explaining how the `withdraw` and `mint` functions operate.